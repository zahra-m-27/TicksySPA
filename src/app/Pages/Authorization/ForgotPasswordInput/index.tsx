import API from '../../../API';
import {useState} from 'react';
import {Button, message} from 'antd';
import styles from './styles.module.scss';
import {useHistory} from 'react-router-dom';
import SEInput from '../../../Components/SEInput';

interface Props {
  className?: string;
}

export default function ForgotPasswordInput({className}: Props) {
  const history = useHistory();
  const [Email, setEmail] = useState('');
  const [Loading, setLoading] = useState(false);
  const [EmailHasError, setEmailHasError] = useState(false);

  const onForgotPassword = () => {
    if (!/^\S+@\S+$/.test(Email)) {
      setEmailHasError(true);
      return;
    }
    setEmailHasError(false);

    setLoading(true);
    API.Users.RequestResetPassword({
      email: Email,
    })
      .then(() => {
        message.success('ایمیل تغییر رمزعبور براتان ارسال شده است', 15);
        history.replace('/');
      })
      .catch((error) => {
        if (error.status === 400) {
          message.error('ایمیل یا رمزعبور اشتباه است');
        }
      })
      .finally(() => setLoading(false));
  };

  return (
    <div className={className}>
      <p className={styles.input_box_title}>فراموشي گذرواژه!</p>

      <SEInput
        type="text"
        label="ايميل"
        regex={/^\S+@\S+$/}
        onChangeText={setEmail}
        hint="example@gmail.com"
        data-testid="email-input"
        hasError={EmailHasError}
        onEnter={onForgotPassword}
        inputClassName={styles.input}
        className={styles.input_class}
        labelClassName={styles.input_label}
        innerContainerClassName={styles.inner_container}
      />

      <Button
        type="primary"
        loading={Loading}
        data-testid="submit-button"
        onClick={onForgotPassword}
        className={styles.enter_button}>
        ثبت
      </Button>
    </div>
  );
}
