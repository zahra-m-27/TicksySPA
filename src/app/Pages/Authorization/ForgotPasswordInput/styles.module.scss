$input_background_color: #fff;

.input_box_title {
  color: #2682cd;
  margin-bottom: 10%;
  align-self: center;
  font-family: IRANMarker;
  font-size: clamp(16px, 2vw, 48px);
}

.input_class {
  width: 79%;
  margin: 0 auto 8%;
}

.input {
  width: 100%;
  font-size: clamp(14px, 1.1vw, 20px);
  background-color: $input_background_color;
}

.input_label {
  font-size: clamp(14px, 1.1vw, 20px);
}

.inner_container {
  background-color: $input_background_color;
}

.enter_button {
  width: 40%;
  height: 42px;
  align-self: center;
}
