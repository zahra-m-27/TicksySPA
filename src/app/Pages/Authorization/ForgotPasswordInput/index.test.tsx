import ReactDOM from 'react-dom';
import React from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import ForgotPasswordInput from './index';
import {MemoryRouter} from 'react-router-dom';

const sample = (
  <MemoryRouter>
    <ForgotPasswordInput />
  </MemoryRouter>
);

test('render in ReactDOM', () => {
  const div = document.createElement('div');
  ReactDOM.render(sample, div);
});

it('render title', () => {
  const {getByText} = render(sample);
  getByText(/فراموشي گذرواژه/i);
});

it('render email field', () => {
  const {getAllByText} = render(sample);
  getAllByText(/ايميل/i);
});

it('render submit button', () => {
  const {getByText} = render(sample);
  getByText(/ثبت/i);
});

it('snapshot', () => {
  const snapshot = renderer.create(sample).toJSON();
  expect(snapshot).toMatchSnapshot();
});
