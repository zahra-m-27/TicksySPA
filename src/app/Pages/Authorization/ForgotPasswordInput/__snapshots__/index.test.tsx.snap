// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <p
    className="input_box_title"
  >
    فراموشي گذرواژه!
  </p>
  <div
    className="input_container input_class"
    data-testid="input-container"
  >
    <div
      className="inner_container inner_container"
    >
      <div
        className="content"
      >
        <p
          className="label input_label"
        >
          ايميل
        </p>
        <input
          className="input input input_label"
          data-testid="email-input"
          dir="auto"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <button
    className="ant-btn ant-btn-primary enter_button"
    data-testid="submit-button"
    onClick={[Function]}
    type="button"
  >
    <span>
      ثبت
    </span>
  </button>
</div>
`;
